src/test/java/com/purbon/kafka/topology/backend/FileBackendTest.java

```diff
@@ -43,9 +43,16 @@ public void after() throws IOException {
 
   @Test
   public void testStoreAndLoadBindingsAndTopics() throws IOException {
-    TopologyAclBinding binding =
-        TopologyAclBinding.build(
-            ResourceType.CLUSTER.name(), "Topic", "host", "op", "principal", "LITERAL");
+    testStoreAndLoadWithPrincipal("principal");
+  }
+
+  @Test
+  public void shouldHandlePrincipalWithSpace() throws IOException {
+    testStoreAndLoadWithPrincipal("User:C=NO,CN=John Doe,emailAddress=john.doe@example.com");
+  }
+
+  private void testStoreAndLoadWithPrincipal(final String principal) throws IOException {
+    TopologyAclBinding binding = TopologyAclBinding.build(ResourceType.CLUSTER.name(), "Topic", "host", "op", principal, "LITERAL");
 
     Topology topology = new TopologyImpl();
```

Comment:
Would be nice to cleanup this old code as well. Can be replaced with:
```
    TestTopologyBuilder builder = TestTopologyBuilder.createProject()
            .addTopic("foo")
            .addTopic("bar");
```

And then e.g. getTopic("foo") on the builder to build the string further down.

Reply:
Please have a new look.

Reply:
LGTM!